### PR TITLE
[test-runner] Allow to define filter inside manifest file

### PR DESCRIPTION
**Pull request by alexshalamov**
Command line option is still available and allows to override filter that is defined in a manifest file.

### CODE REVIEW

File: render-test/manifest_parser.cpp [hunk 1]
```diff
@@ -412,6 +412,18 @@ mbgl::optional<Manifest> ManifestParser::parseManifest(const std::string& manife
         }
     }
 
+    auto filter = testFilter;
```
Comment (pozdnyakov) [inline render-test/manifest_parser.cpp]: instead could be `s/const std::string& testFilter/std::string testFilter/` in the method args

File: render-test/manifest_parser.cpp [hunk 2]
```diff
@@ -412,6 +412,18 @@ mbgl::optional<Manifest> ManifestParser::parseManifest(const std::string& manife
...
+    if (document.HasMember("filter") && filter.empty()) {
```
Comment (pozdnyakov) [inline render-test/manifest_parser.cpp]: nit: ` if ( filter.empty() && document.HasMember("filter"))` would be a bit faster :)

File: render-test/render_test.cpp [hunk 3]
```diff
@@ -101,9 +101,6 @@ ArgumentsTuple parseArguments(int argc, char** argv) {
         mbgl::Log::Info(mbgl::Event::General, stream.str());
         mbgl::Log::Error(mbgl::Event::General, e.what());
         exit(2);
-    } catch (const std::regex_error& e) {
```
Comment (pozdnyakov) [inline render-test/render_test.cpp]: why is this change?

Comment (alexshalamov) [reply render-test/render_test.cpp]: Initially, args::ValueFlag<> filter flag was using regexp as it's type, now, conversion is done in a different place, and `} catch (const std::regex_error& e) {` is never executed.

File: platform/android/src/test/render_test_runner.cpp [hunk 4]
```diff
@@ -252,10 +252,10 @@ void android_main(struct android_app* app) {
     } else {
         unZipFile(env, zipFile, storagePath);
 
-        auto runTestWithManifest = [&storagePath, &app, &outFd, &outEvents, &source](
-                                       const std::string manifest, const std::string filter = {}) -> bool {
+        auto runTestWithManifest =
+            [&storagePath, &app, &outFd, &outEvents, &source](const std::string manifest) -> bool {
```
Comment (pozdnyakov) [inline platform/android/src/test/render_test_runner.cpp]: nit: const std::string& manifest

**Pull request merged**